lib/std/Target.zig

```diff
@@ -2252,23 +2250,20 @@ pub fn c_type_bit_size(target: Target, c_type: CType) u16 {
             .longdouble => return 64,
         },
 
-        .amdhsa, .amdpal => switch (c_type) {
+        .amdhsa, .amdpal, .mesa3d => switch (c_type) {
             .char => return 8,
             .short, .ushort => return 16,
             .int, .uint, .float => return 32,
             .long, .ulong, .longlong, .ulonglong, .double => return 64,
-            .longdouble => return 128,
+            .longdouble => return 64,
         },
 
         .opencl, .vulkan => switch (c_type) {
             .char => return 8,
             .short, .ushort => return 16,
             .int, .uint, .float => return 32,
-            .long, .ulong, .double => return 64,
-            .longlong, .ulonglong => return 128,
-            // Note: The OpenCL specification does not guarantee a particular size for long double,
-            // but clang uses 128 bits.
-            .longdouble => return 128,
+            .long, .ulong, .longlong, .ulonglong, .double => return 64,
```

Comment:
If this is 128 bits for spirv, then this should probably depend on the target, right?

Reply:
You mean for old `spir`? (It's 64 for `spirv`.)

It can't check for old `spir` right now because it was removed in #20870. That's why I mentioned I'll do a follow-up PR to bring that back properly.

Reply:
Ah right, that path is now no longer accessible. Seems fine then.